src/explore-education-statistics-common/src/contexts/NetworkActivityContext.tsx

```diff
@@ -0,0 +1,176 @@
+import useMountedRef from '@common/hooks/useMountedRef';
+import {
+  RequestInterceptor,
+  ResponseInterceptor,
+} from '@common/services/api/Client';
+import React, {
+  createContext,
+  ReactNode,
+  SetStateAction,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+
+const EVENT_NETWORK_REQUEST = 'ees:network_request';
+const EVENT_NETWORK_REQUEST_ERROR = 'ees:network_request_error';
+const EVENT_NETWORK_RESPONSE = 'ees:network_response';
+const EVENT_NETWORK_RESPONSE_ERROR = 'ees:network_response_error';
+
+const dataAttribute = 'data-network-activity';
+
+export type NetworkActivityStatus = 'active' | 'idle';
+
+export interface NetworkActivityState {
+  status: NetworkActivityStatus;
+  requestCount: number;
+}
+
+const NetworkActivityContext = createContext<NetworkActivityState | undefined>(
+  undefined,
+);
+
+export interface NetworkActivityContextProviderProps {
+  children?: ReactNode;
+  idleTimeout?: number;
+}
+
+export function NetworkActivityContextProvider({
```

Comment:
It might be useful to have a comment here explaining why we're doing this as it's not immediately obvious why you'd want to have a data attribute for network activity.